src/server/game-server.ts

```diff
@@ -20,15 +59,31 @@ export class GameServer {
   // The unprocessed inputs for each player
   frameBuffers: {[id:string]: InputFrame[]} = {};
 
-  constructor(sockets: SocketIO.Socket[]) {
+  constructor(gameFinishedCallback: ()=>void) {
+    this.gameFinishedCallback = gameFinishedCallback;
     this.game = Game.init();
     this.lastTick = Date.now();
-    this.sockets = sockets;
+  }
+
+  addPlayer(socket: SocketIO.Socket) {
+    this.sockets.push(socket);
+    this.connectedPlayers += 1;
```

Comment:
Is it not possible to use the length of `this.sockets`?  Would much prefer to use that instead of a separate variable that has to be simultaneously maintained